### ISSUE TITLE
AbstractMethodError with classes extending AbstractSet

### ISSUE THREAD

**Issue by don-vip** [jqno/equalsverifier#161]
# What steps will reproduce the problem?

Test a class extending AbstractSet

# What is the code that triggers this problem?

``` java
import java.util.AbstractSet;
import java.util.Iterator;

public class TestSet extends AbstractSet<Object> {

    private int field;

    @Override
    public Iterator<Object> iterator() {
        return new Iterator<Object>() {
            @Override
            public boolean hasNext() {
                return false;
            }

            @Override
            public Object next() {
                return null;
            }
        };
    }

    @Override
    public int size() {
        return 0;
    }

    @Override
    public int hashCode() {
        return 31 * super.hashCode() + field;
    }

    @Override
    public boolean equals(Object obj) {
        if (this == obj)
            return true;
        if (!super.equals(obj))
            return false;
        if (getClass() != obj.getClass())
            return false;
        TestSet other = (TestSet) obj;
        if (field != other.field)
            return false;
        return true;
    }
}
```

``` java
    @Test
    public void testEqualsContract() {
        EqualsVerifier.forClass(TestSet.class).usingGetClass().verify();
    }
```

# What error message or stack trace does EqualsVerifier give?

```
java.lang.AssertionError: AbstractMethodError: 
For more information, go to: http://www.jqno.nl/equalsverifier/errormessages
	at nl.jqno.equalsverifier.EqualsVerifier.handleError(EqualsVerifier.java:381)
	at nl.jqno.equalsverifier.EqualsVerifier.verify(EqualsVerifier.java:370)
	at org.openstreetmap.josm.data.osm.StorageTest.testEqualsContract2(StorageTest.java:29)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.openstreetmap.josm.testutils.JOSMTestRules$TimeoutThread.run(JOSMTestRules.java:348)
Caused by: java.lang.AbstractMethodError
	at java.util.AbstractSet.equals(AbstractSet.java:92)
	at org.openstreetmap.josm.data.osm.TestSet.equals(TestSet.java:40)
	at nl.jqno.equalsverifier.HierarchyChecker.checkSuperclass(HierarchyChecker.java:77)
	at nl.jqno.equalsverifier.HierarchyChecker.check(HierarchyChecker.java:56)
	at nl.jqno.equalsverifier.EqualsVerifier.verifyWithExamples(EqualsVerifier.java:431)
	at nl.jqno.equalsverifier.EqualsVerifier.performVerification(EqualsVerifier.java:393)
	at nl.jqno.equalsverifier.EqualsVerifier.verify(EqualsVerifier.java:364)
	... 10 more
```

# Which version of EqualsVerifier are you using?

2.1.7

# Please provide any additional information below.

This comes probably from the fact that AsbtractSet does not define `size()`

**Comment by jqno** [comment]
I've reproduced the issue, and you're right, it's because `AbstractSet` doesn't define `size()`.

I have a solution in mind for this, but you caught me in a busy period so it might be a little while before I get to it, unfortunately.

**Comment by jqno** [comment]
I just released version 2.1.8 which contains a fix for this.
